virtualcluster/cmd/syncer/app/server.go

```diff
@@ -132,6 +132,22 @@ func Run(cc *syncerconfig.CompletedConfig, stopCh <-chan struct{}) error {
 		}
 	}()
 
+	go func() {
+		// start a pprof http server
+		klog.Fatal(http.ListenAndServe(":6060", nil))
+	}()
+
+	go func() {
+		// start a health http server.
+		mux := http.NewServeMux()
+		healthz.InstallHandler(mux)
+		klog.Fatal(http.ListenAndServe(":8080", mux))
+	}()
+
+	go func() {
+		ss.ListenAndServe(net.JoinHostPort(cc.Address, cc.Port), cc.CertFile, cc.KeyFile)
```

Comment:
Aha, digging into the syncer code, it seems like this is actually for *metrics* serving rather than for webhooks: https://github.com/kubernetes-sigs/cluster-api-provider-nested/blob/e5ca44baf3040c0f5cbeed7b20f2fd3c2c0b4d72/virtualcluster/pkg/syncer/syncer.go#L261

Given that the syncer isn't running when not elected (`s.Run` has not been called), I think this will just be an empty page of metrics which is what we want & shouldn't require any additional work in prometheus rules...

Reply:
As the whole work will be done in the leader, having metrics from the non-leader instance is not a big deal, as they are eventually reset